Islandora-CLAW/islandora
# Rules actions for event message

Resolves https://github.com/Islandora-CLAW/CLAW/issues/450
See the associated install PR https://github.com/Islandora-CLAW/CLAW/pull/451

This PR has three rules actions, one to create each type of event message.  It packages those actions into three reaction rules.  Since https://github.com/Islandora-CLAW/CLAW/issues/448 and https://github.com/Islandora-CLAW/CLAW/issues/432 are still on the go, we simple print the event messages to the screen instead actually doing any queue publishing.

It also comes with more Kernel tests for the actions, along with a healthy refactoring of the tests so phpcpd doesn't complain and PHPUnit has info on code coverage.

### To test
- Uninstall islandora
- Install `typed_data` and `rules` modules
- Pull down this PR
- Re-install islandora
- Verify the rules have installed themselves by visiting `admin/config/workflow/rules`
![screenshot from 2016-12-09 15-54-58](https://cloud.githubusercontent.com/assets/20773151/21062857/1c10fc7a-be29-11e6-9238-438a880c2e7e.png)
- Go to edit the rules to verify there are two actions, one for generating the message and another to display it
![screenshot from 2016-12-09 15-55-39](https://cloud.githubusercontent.com/assets/20773151/21062880/3a9d4ac2-be29-11e6-9acc-86673280b973.png)
- Then go create a FedoraResource, and a create event message should be dsm'd to the screen
![screenshot from 2016-12-09 15-56-31](https://cloud.githubusercontent.com/assets/20773151/21062900/58b8afc4-be29-11e6-83ec-3d2f4502e716.png)
- Edit the entity, and you should see an update event message
![screenshot from 2016-12-09 15-57-16](https://cloud.githubusercontent.com/assets/20773151/21062912/6b4cf7ee-be29-11e6-845b-a068874d7499.png)
- Delete the entity and you should see a delete event message
![screenshot from 2016-12-09 15-57-45](https://cloud.githubusercontent.com/assets/20773151/21062926/76312b62-be29-11e6-8c58-7b181b13b462.png)
